tutorials/graphql/intro-graphql/tutorial-site/content/graphql-vs-grpc.md

```diff
@@ -0,0 +1,88 @@
+
+Consider the following example:
+
+```
+syntax = "proto3";
+
+service User {
+  rpc GetUser (UserRequest) returns (UserResponse) {}
+}
+
+message UserRequest {
+  int32 id = 1;
+}
+
+message UserResponse {
+  string name = 1;
+  int32 age = 1;
+  string address = 1;
+}
+```
+
+The above code defines a service that allows you to retrieve the details about a specific user. The `GetUser` method represents the API endpoint the client can call to retrieve a user.
+
+Due to the `.proto` files and Protocol buffers, gRPC can auto-generate the client & server boilerplate code for yourself in various programming languages.
+
+## gRPC Benefits
+
+One benefit of using gRPC is the performance of data exchange. gRPC uses Protobuf to serialize data into binary format, which improves the payload size. As a result, exchanging data is faster and more efficient.
+
+gRPC also uses the HTTP/2 transport protocol, which comes with request and response multiplexing. In HTTP/2, the request/response is split into multiple frames that are sent individually and put together back when they arrive at the destination. That makes it possible to transfer multiple requests and responses in one connection.
+
+Another advantage of using gRPC is that it supports various streaming types. It supports
+* Server streaming - the client makes a request, and the server responds with a stream of messages
+* Client streaming - the client sends a stream of messages to the server, which responds after the client has finished streaming
+* Bidirectional streaming - both the client and server send independent streams of messages to each other
+
+gRPC also supports Unary interactions, where the client sends one request, and the server sends one response back.
+
+Code generation is another benefit. gRPC's `protoc` compiler can use the `.proto` file to generate server and client code in 11 programming languages.
+
+## gRPC Drawbacks
+
+One of the main drawbacks of gRPC is that it does not come with browser support. That means a browser can't communicate with a gRPC service. There are some workarounds to do it using HTTP/1.1, but you lose the benefits of HTTP/2.
+
+Another drawback is that Protobuf (Protocol buffers), a core part of gRPC, only supports code generated in 9 languages: Kotlin, Python, C++, Ruby, Java, C#, Go, Objective-C, and Dart.
+
+Since gRPC uses Protobuf to exchange data, the messages are not human readable. Reading and inspecting data requires extra steps and tools.
+
+## GraphQL vs gRPC
+
+When it comes to data fetching, GraphQL is more precise than gRPC. That means you can retrieve exactly the data you want - nothing more, nothing less. With gRPC, you might get extra data back from the server (similar to REST).
```

Comment:
gRPC is similar to REST in that the API might be designed to over-fetch data in some cases.